cfg/aws/aws_sdk_logging_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/influxdata/wlog"
+)
+
+func TestSetSDKLogLevel(t *testing.T) {
+	cases := []struct {
+		wLogLevel         wlog.Level
+		sdkLogLevelString string
+		expectedVal       aws.LogLevelType
+	}{
+		// ENV VAR does not match
+		{wlog.DEBUG, "FOO", aws.LogOff},
+		{wlog.INFO, "FOO", aws.LogOff},
+		{wlog.WARN, "FOO", aws.LogOff},
+		{wlog.ERROR, "FOO", aws.LogOff},
+		// ENV VAR matches, but wLogLevel is not DEBUG.
+		{wlog.INFO, "DEBUG", aws.LogOff},
+		{wlog.INFO, "DEBUG", aws.LogOff},
+		{wlog.WARN, "DEBUG", aws.LogOff},
+		{wlog.ERROR, "DEBUG", aws.LogOff},
+		// ENV VAR matches, wLogLevel is DEBUG.
+		{wlog.DEBUG, "LogDebug", aws.LogDebug},
+		{wlog.DEBUG, "LogDebugWithEventStreamBody", aws.LogDebugWithEventStreamBody},
+		{wlog.DEBUG, "LogDebugWithHTTPBody", aws.LogDebugWithHTTPBody},
+		{wlog.DEBUG, "LogDebugWithSigning", aws.LogDebugWithSigning},
+		// Multiple matches
+		{wlog.DEBUG, "LogDebugWithEventStreamBody|LogDebugWithSigning",
```

Comment:
Okay, so now I'm confused. I had assumed that the log level was going to be a single string, e.g.: `LogDebugWithHTTPBody`. This implies that we can supply a `|` delimited string for the log level. Would appreciate if this was explained more.

Reply:
SDK src code:
https://github.com/aws/aws-sdk-go/blob/v1.41.18/aws/logger.go#L10
Basically the various log levels are a bitwise field.
LogOff = 0x00
LogDebug = 0x1000
LogDebugWithSigning = 0x1001
LogDebugWithHTTPBody = 0x1002
LogDebugWithRequestRetries = 0x1004

I assumed since the the `LogLevelType.Matches()` method does bitwise comparison, it is valid to set multiple bits:
```
// Matches returns true if the v LogLevel is enabled by this LogLevel. Should be
// used with logging sub levels. Is safe to use on nil value LogLevelTypes. If
// LogLevel is nil, will default to LogOff comparison.
func (l *LogLevelType) Matches(v LogLevelType) bool {
	c := l.Value()
	return c&v == v
}
```

To be honest the SDK could be better about explaining with examples using various LogLevels.
I did find this:
https://sourcegraph.com/github.com/aws/aws-sdk-go/-/docs/aws#LogDebugWithHTTPBody

Even if the SDK allows users to combine levels, we don't necessarily need to allow it via the CloudWatch Agent.
If we think it is useful enough to just allow users to set `"LogDebugWithHTTPBody"` or `"LogDebugWithSigning"` then I am fine with changing the code to just choose the last matching string on the line.

Reply:
I think it's fine leaving this as is, though it'd be nice to know what happens when you mix and match them - maybe not necessarily our documentation issue to solve.